src/documents/migrations/1053_document_pages_count.py

```diff
@@ -0,0 +1,71 @@
+# Generated by Django 4.2.16 on 2024-09-21 15:44
+import os
+
+import pikepdf
+from django.conf import settings
+from django.db import migrations
+from django.db import models
+from django.utils.termcolors import colorize as colourise
+
+
+def source_path(self):
+    if self.filename:
+        fname = str(self.filename)
+
+    return os.path.join(settings.ORIGINALS_DIR, fname)
+
+
+def add_number_of_pages_to_pages_count(apps, schema_editor):
+    Document = apps.get_model("documents", "Document")
+
+    if not Document.objects.all().exists():
+        return
+
+    for doc in Document.objects.filter(mime_type="application/pdf"):
+        print(
+            "    {} {} {}".format(
+                colourise("*", fg="green"),
+                colourise("Calculating number of pages for", fg="white"),
+                colourise(doc.filename, fg="cyan"),
+            ),
+        )
+
+        pdf = pikepdf.open(source_path(doc))
+
+        if pdf.pages is not None:
+            doc.pages_count = len(pdf.pages)
+            doc.save()
```

Comment:
```suggestion
        try:
            pdf = pikepdf.open(document.source_path)

            if pdf.pages is not None:
                doc.pages_count = len(pdf.pages)
                doc.save()
        except Exception as e:
            print(f"Error retrieving number of pages for {document.filename}: {e}")
```

I think this got lost, again basic error handling during migration to prevent boot loop

Reply:
> I think this got lost, again basic error handling during migration to prevent boot loop

It's strange, I thought I had already committed the first time you suggested this change...
But thanks